app/controllers/add-preprint.js

```diff
@@ -95,55 +95,61 @@ export default Ember.Controller.extend(Validations, NodeActionsMixin, {
     }),
 
     /*
-     * Subjects section
-     */
-    filter: [{/* value: 'filterQuery' */}, {}, {}],
-    filteredPath: Ember.computed('path', 'filter', 'filter.@each.value', function() {
-        return this.get('path').slice(0, 2).map((path, i) => {
-            if (path.children && this.get(`filter.${i + 1}.value`)) {
-                return {
-                    name: path.name,
-                    children: path.children.filter(child =>
-                        this.get(`filter.${i + 1}.value`).indexOf(child.name || child) !== -1)
-                };
-            }
-            return path;
-        });
+    * Subjects section
+    */
+    topFilter: '',
+    midFilter: '',
+    botFilter: '',
```

Comment:
Will every taxonomy always have exactly three levels of filtering (and no more)?